File: ontrack-extension-git/src/main/java/net/nemerosa/ontrack/extension/git/model/GitChangeLogCommits.java
package net.nemerosa.ontrack.extension.git.model;

import lombok.Data;
import net.nemerosa.ontrack.extension.scm.model.SCMChangeLogCommit;
import net.nemerosa.ontrack.extension.scm.model.SCMChangeLogCommits;

import java.util.List;

@Data
public class GitChangeLogCommits implements SCMChangeLogCommits {

    private final GitUILog log;

    @Override
    public List<? extends SCMChangeLogCommit> getCommits() {
        return log.getCommits();
    }
}

File: ontrack-extension-git/src/main/java/net/nemerosa/ontrack/extension/git/model/GitChangeLogCommits.kt
package net.nemerosa.ontrack.extension.git.model

import net.nemerosa.ontrack.extension.scm.model.SCMChangeLogCommit
import net.nemerosa.ontrack.extension.scm.model.SCMChangeLogCommits

// TODO #532 Workaround
open class GitChangeLogCommits(
    val log: GitUILog
) : SCMChangeLogCommits {

    override val commits: List<SCMChangeLogCommit> = log.commits

}
